src/ExpressionLanguage/ExpressionFunction/GraphQL/Relay/GlobalID.php

```diff
@@ -5,23 +5,29 @@
 namespace Overblog\GraphQLBundle\ExpressionLanguage\ExpressionFunction\GraphQL\Relay;
 
 use Overblog\GraphQLBundle\ExpressionLanguage\ExpressionFunction;
+use Overblog\GraphQLBundle\Relay\Node\GlobalId as GlobalIdHelper;
 
 final class GlobalID extends ExpressionFunction
 {
     public function __construct($name = 'globalId')
     {
         parent::__construct(
             $name,
-            function ($id, $typeName = null) {
-                $typeNameEmpty = null === $typeName || '""' === $typeName || 'null' === $typeName || 'false' === $typeName;
+            function (string $id, string $typeName = null): string {
+                $typeName = $this->isTypeNameEmpty($typeName) ? '$info->parentType->name' : $typeName;
 
-                return \sprintf(
-                    '\%s::toGlobalId(%s, %s)',
-                    \Overblog\GraphQLBundle\Relay\Node\GlobalId::class,
-                    \sprintf($typeNameEmpty ? '$info->parentType->name' : '%s', $typeName),
-                    $id
-                );
+                return \sprintf('\%s::toGlobalId(%s, %s)', GlobalIdHelper::class, $typeName, $id);
+            },
+            function ($arguments, $id, $typeName = null) {
+                $typeName = $this->isTypeNameEmpty($typeName) ? $arguments['info']->parentType->name : $typeName;
```

Comment:
```suggestion
                $typeName = empty($typeName) ? $arguments['info']->parentType->name : $typeName;
```

Reply:
Why replaced with `empty`? I just used the code from the compiler callback

Reply:
compiler callback is for code string, evaluator is for php values (`boolean`, `string`, ...) so `empty` does the because it also cover `false` vs `isTypeNameEmpty` method that cover `'false'`.

Reply:
Fixed!